Use .ifexists meta command to load optional modules only if available

From: CJ van den Berg <cj@vdbonline.com>


---

 src/daemon/default.pa.in |   18 +++++++++++++++---
 1 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index af2a678..b769e3f 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -26,12 +26,18 @@
 #load-module module-null-sink
 #load-module module-pipe-sink
 
+.ifexists /usr/lib/pulse-0.9/modules/module-hal-detect.so
+
 ### Automatically load driver modules depending on the hardware available
-@HAVE_HAL_TRUE@load-module module-hal-detect
+load-module module-hal-detect
+
+.else
 
 ### Alternatively use the static hardware detection module (for systems that
 ### lack HAL support
-@HAVE_HAL_FALSE@load-module module-detect
+load-module module-detect
+
+.endif
 
 ### Load audio drivers automatically on access
 #add-autoload-sink output module-oss device="/dev/dsp" sink_name=output source_name=input
@@ -41,8 +47,14 @@
 #add-autoload-sink output module-alsa-sink sink_name=output
 #add-autoload-source input module-alsa-source source_name=input
 
-### Load several protocols
+.ifexists /usr/lib/pulse-0.9/modules/module-esound-protocol-unix.so
+
+### Load esound protocol
 load-module module-esound-protocol-unix
+
+.endif
+
+### Load native protocol
 load-module module-native-protocol-unix
 
 ### Network access (may be configured with paprefs, so leave this commented
